main_test.go

```diff
@@ -583,37 +573,49 @@ func makeEventResponse(success bool) *pbevent.Response {
 	return &pbevent.Response{Status: status, Message: msg}
 }
 
-func (s *eventTestServer) NotifyCollation(
+func (s *eventTestServer) Receive(
 	ctx context.Context,
-	req *pbevent.NotifyCollationRequest) (*pbevent.NotifyCollationResponse, error) {
+	req *pbevent.ReceiveRequest) (*pbevent.ReceiveResponse, error) {
 	go func() {
-		s.collations <- req.Collation
+		s.receivedData <- req.Data
 	}()
-	res := &pbevent.NotifyCollationResponse{
-		Response: makeEventResponse(true),
-		IsValid:  true,
-	}
-	return res, nil
-}
-func (s *eventTestServer) GetCollation(
-	ctx context.Context,
-	req *pbevent.GetCollationRequest) (*pbevent.GetCollationResponse, error) {
-	var collation *pbmsg.Collation
-	// simulate the case when collation is not found
-	if int(req.ShardID) == 1 && int(req.Period) == 42 {
-		collation = nil
-	} else {
-		collation = &pbmsg.Collation{
-			ShardID: req.ShardID,
-			Period:  req.Period,
-			Blobs:   []byte(fmt.Sprintf("shardID=%v, period=%v", req.ShardID, req.Period)),
+	var resBytes []byte
+	success := true
+	switch int(req.MsgType) {
+	case typeCollation:
+		msg := &pbmsg.Collation{}
+		err := proto.Unmarshal(req.Data, msg)
+		if err != nil {
+			success = false
 		}
+		resBytes = []byte{1} // true
+	case typeCollationRequest:
+		msg := &pbmsg.CollationRequest{}
+		err := proto.Unmarshal(req.Data, msg)
+		if err != nil {
+			success = false
+		}
+		if msg.ShardID != 42 {
+			collation := &pbmsg.Collation{
+				ShardID: msg.ShardID,
+				Period:  msg.Period,
+				Blobs:   []byte(fmt.Sprintf("%v:%v:blobssss", msg.ShardID, msg.Period)),
+			}
+			resBytes, err = proto.Marshal(collation)
+			if err != nil {
+				success = false
+			}
+		} else {
+			success = false
+			// resBytes = []byte{} // return empty bytes to indicate that we don't have the collation
+		}
+	default:
+		success = false
```

Comment:
init `success` true and default here false is weird

Reply:
Oh ... I see how it works

Reply:
> init success true and default here false is weird

You're right I think. Will it be better if we just declare it without initializing to a specific value? like`var success bool`. I.e. it implies `success` is initialized to `false`